### PR TITLE
STL-1677 Fahrkostenpauschale Step

### PR DESCRIPTION
**Pull request by SannyNguyenHung**
# Short Description
- Add Fahrkostenpauschale step and make it available in the steuerlotse app.
- I merged some changes from other branches, like preconditions or `get_overview_value_representation`

# Feedback
- Do I miss something?
- Any!

# How to review this Pull Request
[Link to Confluence](https://digitalservice4germany.atlassian.net/wiki/spaces/STL/pages/117637157/Development+Process#Approve-or-comment) (internal)

### CODE REVIEW

File: webapp/app/forms/flows/lotse_flow.py
Comment (SannyNguyenHung) [inline webapp/app/forms/flows/lotse_flow.py:363]: If the function get_overview_value_representation() returns `None`, the value will not be displayed.

File: webapp/app/forms/steps/lotse/personal_data.py
Comment (SannyNguyenHung) [inline webapp/app/forms/steps/lotse/personal_data.py:551]: I know this is in PR https://github.com/digitalservice4germany/steuerlotse/pull/516 but I need it here also.

File: webapp/app/forms/flows/lotse_flow.py
Comment (JulianRoesner) [reply webapp/app/forms/flows/lotse_flow.py:363]: Maybe you could add this as a comment in the code?

Comment (JulianRoesner) [reply webapp/app/forms/flows/lotse_flow.py:363]: I think we should again think more about how we combine this part with the `_generate_value_representation`. However, this could also be a task for the refactoring of the steps

File: webapp/app/forms/steps/lotse/fahrkostenpauschale.py
Comment (JulianRoesner) [inline webapp/app/forms/steps/lotse/fahrkostenpauschale.py:33]: Shouldn't we add an abstract `get_fahrkostenpauschale` method? If we do so, we should probably add one to the PauschbetragStep, too.

File: webapp/app/forms/steps/lotse/fahrkostenpauschale.py
Comment (JulianRoesner) [inline webapp/app/forms/steps/lotse/fahrkostenpauschale.py:75]: If #532 is merged before this one, we could get rid of this part

File: webapp/app/forms/steps/lotse/fahrkostenpauschale.py
Comment (JulianRoesner) [inline webapp/app/forms/steps/lotse/fahrkostenpauschale.py:132]: Same here

File: webapp/app/forms/steps/lotse/personal_data.py
Comment (JulianRoesner) [reply webapp/app/forms/steps/lotse/personal_data.py:551]: 👍 When merging with main, we will need to make sure that this precondition is replaced with the correct one (in` has_disability.py`)

File: webapp/app/forms/steps/step.py
Comment (JulianRoesner) [inline webapp/app/forms/steps/step.py:60]: This was also already added in another PR, correct?

File: webapp/app/translations/de/LC_MESSAGES/messages.po
Comment (JulianRoesner) [inline webapp/app/translations/de/LC_MESSAGES/messages.po:1176]: Shouldn't this be used in more places than just once?

File: webapp/app/translations/de/LC_MESSAGES/messages.po
Comment (JulianRoesner) [inline webapp/app/translations/de/LC_MESSAGES/messages.po:5371]: We should make sure that these messages not get deleted when merging with main

Comment (JulianRoesner) [reply webapp/app/translations/de/LC_MESSAGES/messages.po:5371]: Maybe just delete those lines before merging

File: webapp/client/cypress/integration/personAHasDisability.spec.js
Comment (JulianRoesner) [inline webapp/client/cypress/integration/personAHasDisability.spec.js:39]: When merging this probably has to be changed to the way it already is on main and the pauschbetrag functional tests have to include something like this test.

File: webapp/client/cypress/integration/personBHasDisability.spec.js
Comment (JulianRoesner) [inline webapp/client/cypress/integration/personBHasDisability.spec.js:47]: Same as above for the person a functional tests

File: webapp/client/src/pages/FahrkostenpauschalePersonAPage.js
Comment (JulianRoesner) [inline webapp/client/src/pages/FahrkostenpauschalePersonAPage.js:35]: Why did you have to change this? Shouldn't we then also have to change the fieldType in `FahrkostenpauschalePage.js`?

File: webapp/app/forms/steps/lotse/fahrkostenpauschale.py
Comment (JulianRoesner) [inline webapp/app/forms/steps/lotse/fahrkostenpauschale.py:24]: Why don't we return None but 0?

File: webapp/tests/forms/steps/lotse/test_fahrkostenpauschbetrag_steps.py
Comment (JulianRoesner) [inline webapp/tests/forms/steps/lotse/test_fahrkostenpauschbetrag_steps.py:19]: If we add this to the functional tests, we do not need to test this here

File: webapp/tests/forms/steps/lotse/test_fahrkostenpauschbetrag_steps.py
Comment (JulianRoesner) [inline webapp/tests/forms/steps/lotse/test_fahrkostenpauschbetrag_steps.py:12]: could we also add a test for a failed validation?

File: webapp/tests/forms/steps/lotse/test_fahrkostenpauschbetrag_steps.py
Comment (JulianRoesner) [inline webapp/tests/forms/steps/lotse/test_fahrkostenpauschbetrag_steps.py:60]: I think these two both could also be functional tests. WDYT?

File: webapp/tests/forms/steps/lotse/test_fahrkostenpauschbetrag_steps.py
Comment (JulianRoesner) [inline webapp/tests/forms/steps/lotse/test_fahrkostenpauschbetrag_steps.py:36]: Same as above: Could we add a test for a failed validation?

**Review by JulianRoesner — CHANGES_REQUESTED** [review]
It looks good in general. I think there are some things that need to be adapted when main has been merged. 

Besides that, could you add functional tests for the Fahrkostenpauschale Step? I think they should be quite similar to the ones in #547

File: webapp/app/forms/steps/step.py
Comment (JulianRoesner) [reply webapp/app/forms/steps/step.py:60]: Yes. However, it was a class method and it is now an instance method.

File: webapp/app/translations/de/LC_MESSAGES/messages.po
Comment (JulianRoesner) [reply webapp/app/translations/de/LC_MESSAGES/messages.po:1176]: It is used here and in the pauschbetrag step. However, this has not been merged.

File: webapp/app/forms/flows/lotse_flow.py
Comment (JulianRoesner) [reply webapp/app/forms/flows/lotse_flow.py:363]: On main we already had a ToDo comment for combining `_generate_value_representation`. I added the comment

File: webapp/app/forms/steps/lotse/personal_data.py
Comment (JulianRoesner) [reply webapp/app/forms/steps/lotse/personal_data.py:551]: I checked this.

File: webapp/tests/forms/steps/lotse/test_fahrkostenpauschbetrag_steps.py
Comment (JulianRoesner) [reply webapp/tests/forms/steps/lotse/test_fahrkostenpauschbetrag_steps.py:19]: I added the mentioned functional tests

File: webapp/tests/forms/steps/lotse/test_fahrkostenpauschbetrag_steps.py
Comment (JulianRoesner) [reply webapp/tests/forms/steps/lotse/test_fahrkostenpauschbetrag_steps.py:12]: I added a test for this

File: webapp/tests/forms/steps/lotse/test_fahrkostenpauschbetrag_steps.py
Comment (JulianRoesner) [reply webapp/tests/forms/steps/lotse/test_fahrkostenpauschbetrag_steps.py:36]: I added a test for this

File: webapp/app/forms/steps/lotse/fahrkostenpauschale.py
Comment (SannyNguyenHung) [reply webapp/app/forms/steps/lotse/fahrkostenpauschale.py:28]: When we render the value the text could be: "None Euro" - I know this is maybe not a possible case but just in case :grin:

File: webapp/app/forms/steps/lotse/fahrkostenpauschale.py
Comment (SannyNguyenHung) [reply webapp/app/forms/steps/lotse/fahrkostenpauschale.py:37]: :+1:

File: webapp/app/forms/steps/lotse/fahrkostenpauschale.py
Comment (SannyNguyenHung) [inline webapp/app/forms/steps/lotse/fahrkostenpauschale.py:93]: This can't work... person_a_has_pflegegrad is a 'yes, 'no' string

File: webapp/app/forms/steps/lotse/fahrkostenpauschale.py
Comment (SannyNguyenHung) [inline webapp/app/forms/steps/lotse/fahrkostenpauschale.py:145]: Same like person_a_has_pflegegrad

**Review by SannyNguyenHung — COMMENTED** [review]
Please check person_a_has_pflegegrad & person_b_has_pflegegrad. Those are 'yes', 'no' string fields.

File: webapp/app/forms/steps/lotse/fahrkostenpauschale.py
Comment (JulianRoesner) [reply webapp/app/forms/steps/lotse/fahrkostenpauschale.py:93]: Valid point. This is changed in #538 . We should maybe just take that implementation here.

**Review by SannyNguyenHung — COMMENTED** [review]
I can't approve this PR :grin: but we can merge it!

**Review by JulianRoesner — APPROVED** [review]
🚀

**Pull request merged**
